### PR TITLE
feat: completed cart management with price details

### PR DESCRIPTION
**Pull request by ajit-sharmaGH**
image
image

### CODE REVIEW

File: src/components/cart/cart.jsx [hunk 1]
```diff
@@ -1,67 +1,138 @@
...
+  const { cartState, cartItemsDispatch } = useCart();
+  const totalCalculatedCost = () => {
+    return cartState.cartData.reduce((sum, currentPrice) => {
+      return (sum = sum + currentPrice.totalPrice);
+    }, 0);
+  };
+  const discountCalculatedCost = () => {
+    return cartState.cartData.reduce((sum, currentValue) => {
+      return (sum = sum + currentValue.discountPrice * currentValue.quantity);
+    }, 0);
```
Comment (t007rushi) [inline src/components/cart/cart.jsx:13]: you can optimize with a single reduce just suggestion

File: src/components/cart/cart.jsx [hunk 2]
```diff
@@ -1,67 +1,138 @@
...
+                <li>
+                  Rs:
+                  {`${totalCalculatedCost() - discountCalculatedCost() + 100}`}
+                </li>
+              </div>
+              <hr className="hr-tag" />
+              <div className="flex-row delivery-charge">
+                <li className="price-tag">
+                  You Will Save: Rs:&nbsp;{`${discountCalculatedCost()}`}{" "}
+                  &nbsp;on this order
```
Comment (t007rushi) [inline src/components/cart/cart.jsx:123]: use icons instead

File: src/components/products/productsCard.jsx [hunk 3]
```diff
@@ -34,7 +34,30 @@ const ProductsCard = () => {
...
+      payload: {
+        value: {
+          _id,
+          title,
+          productImage,
+          discountPrice,
+          originalPrice,
+          quantity: 1,
+          totalPrice: originalPrice,
+          rating,
```
Comment (t007rushi) [inline src/components/products/productsCard.jsx]: is there need to destructure again can use ```...rest``` instead

File: src/reducers/cart-reducer.js [hunk 4]
```diff
@@ -1,10 +1,50 @@
...
+
+        case "DECREASE_QUANTITY":
+            return {
+                ...state,
+                cartData: state.cartData.map((data) =>
+                    data._id === action.payload.value ?
+                    {
+                        ...data,
+                        quantity: data.quantity - 1,
+                        totalPrice: (data.quantity - 1) * data.originalPrice,
```
Comment (t007rushi) [inline src/reducers/cart-reducer.js:27]: this calculations must be separated out as easy to get calculated when needed in reduce itself

File: src/reducers/cart-reducer.js [hunk 5]
```diff
@@ -1,10 +1,50 @@
 const cartReducerFun = (state, action) => {
     switch (action.type) {
         case "ADD_TO_CART":
-            return {...state, cartCounter: state.cartCounter + 1, }
+            return {
+                ...state,
+                cartData: [...state.cartData, action.payload.value],
+                cartCounter: state.cartCounter + 1,
```
Comment (t007rushi) [inline src/reducers/cart-reducer.js:7]: you can count cart items using ```.length`` method

**Review by t007rushi — APPROVED** [review]
You can look for better contrast colors for filters that look more delightful to the eyes just suggestion

File: src/components/products/productsCard.jsx (hunk 3 above)
Comment (ajit-sharmaGH) [reply src/components/products/productsCard.jsx]: _yes yes  will try that to improve thanks captain for pointing it out_

**Review by Mohasin-Haque — COMMENTED** [review]
Your code is good and your folder structure also. You can work on the mentioned suggestions.

**Pull request merged**